From 91e3ea59acb6de24d02eaad50df8f00a05f4d80d Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas@debian.org>
Date: Fri, 1 Jan 2021 15:38:40 +0100
Subject: [PATCH 08/21] Simplify generation of temporary i586-* symlinks for
 stretch

Make tests and variables produce more explicit logs.

Simplify two reciprocal transformations:
compat := CROSS_GNU_TYPE with s/i686/i586/
CROSS_GNU_TYPE = _gnu_type with s/i586/i686/
---
 debian/rules | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/debian/rules b/debian/rules
index 7c14e34..87417ad 100755
--- a/debian/rules
+++ b/debian/rules
@@ -953,14 +953,13 @@ stamps/install.%: stamps/build.%
 	chrpath -k -d $$(file $(D_CROSS)/usr/bin/*|awk -F: '/ ELF/ {print $$1}')
 
 ifneq (,$(filter $(distrelease),lenny etch squeeze wheezy jessie stretch))
+  ifneq (,$(filter i686-%,$(CROSS_GNU_TYPE)))
 	: # temporary i586-* symlinks for stretch
-	compat=$$(echo $(CROSS_GNU_TYPE) | sed 's/i686/i586/'); \
-	case "$(CROSS_GNU_TYPE)" in i686-*) \
-	  for i in $(programs); do \
-	    ln -sf $(CROSS_GNU_TYPE)-$$i $(D_CROSS)/$(PF)/bin/$$compat-$$i; \
-	    ln -sf $(CROSS_GNU_TYPE)-$$i.1.gz $(D_CROSS)/$(PF)/share/man/man1/$$compat-$$i.1.gz; \
-	  done; \
-	esac
+	for i in $(programs); do \
+	  ln -sf $(CROSS_GNU_TYPE)-$$i $(D_CROSS)/$(PF)/bin/$(call _gnu_type,$(CROSS_ARCH))-$$i; \
+	  ln -sf $(CROSS_GNU_TYPE)-$$i.1.gz $(D_CROSS)/$(PF)/share/man/man1/$(call _gnu_type,$(CROSS_ARCH))-$$i.1.gz; \
+	done
+  endif
 endif
 	touch $@
 
-- 
2.30.1

